Please indicate which PhpStan level your changes allow to pass and for which files. It is almost good for level 9:

```console
$ vendor/bin/phpstan analyse --level 9 app/Models/Days.php  app/Utils/feverUtil.php app/Utils/passwordUtil.php 
Note: Using configuration file /home/alex/GitHub/FreshRSS/phpstan.neon.
 3/3 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ -------------------------------------------------------------------- 
  Line   Utils/feverUtil.php                                                 
 ------ -------------------------------------------------------------------- 
  :29    Cannot access property $salt on FreshRSS_SystemConfiguration|null.  
 ------ -------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------- 
  Line   Utils/passwordUtil.php                                                               
 ------ ------------------------------------------------------------------------------------- 
  :23    Method FreshRSS_password_Util::hash() should return string but returns string|null.  
 ------ ------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 2 errors
```

--- a/app/Utils/feverUtil.php
+++ b/app/Utils/feverUtil.php
@@ -14,7 +14,7 @@ class FreshRSS_fever_Util {
 		}
 
 		$isOk = touch(self::FEVER_PATH . '/index.html');	// is_writable() is not reliable for a folder on NFS
-		if (false === $isOk) {
+		if ($isOk === false) {
 			Minz_Log::error("Could not save Fever API credentials. The directory does not have write access.");
 		}
 		return $isOk;

--- a/app/Utils/feverUtil.php
+++ b/app/Utils/feverUtil.php
@@ -13,17 +13,20 @@ class FreshRSS_fever_Util {
 			@mkdir(self::FEVER_PATH, 0770, true);
 		}
 
-		$isOk = touch(self::FEVER_PATH . '/index.html');	// is_writable() is not reliable for a folder on NFS
-		if ($isOk === false) {
+		$ok = touch(self::FEVER_PATH . '/index.html');	// is_writable() is not reliable for a folder on NFS
+		if ($ok) {
 			Minz_Log::error("Could not save Fever API credentials. The directory does not have write access.");
 		}
-		return $isOk;
+		return $ok;
 	}
 
 	/**
 	 * Return the corresponding path for a fever key.
 	 */
 	public static function getKeyPath(string $feverKey): string {
+		if (FreshRSS_Context::$system_conf === null) {
+			throw new FreshRSS_Context_Exception('System configuration not initialised!');
+		}
 		$salt = sha1(FreshRSS_Context::$system_conf->salt);
 		return self::FEVER_PATH . '/.key-' . $salt . '-' . $feverKey . '.txt';
 	}
@@ -33,7 +36,7 @@ class FreshRSS_fever_Util {
 	 * @return string|false the Fever key, or false if the update failed
 	 */
 	public static function updateKey(string $username, string $passwordPlain) {
-		if (self::checkFeverPath() === false) {
+		if (!self::checkFeverPath()) {
 			return false;
 		}
 
@@ -42,7 +45,7 @@ class FreshRSS_fever_Util {
 		$feverKey = strtolower(md5("{$username}:{$passwordPlain}"));
 		$feverKeyPath = self::getKeyPath($feverKey);
 		$result = file_put_contents($feverKeyPath, $username);
-		if (is_int($result) === true) {
+		if (is_int($result) && $result > 0) {
 			return $feverKey;
 		}
 		Minz_Log::warning('Could not save Fever API credentials. Unknown error.', ADMIN_LOG);

--- a/app/Utils/feverUtil.php
+++ b/app/Utils/feverUtil.php
@@ -1,7 +1,5 @@
 <?php
 
-declare(strict_types=1);
-
 class FreshRSS_fever_Util {
 	private const FEVER_PATH = DATA_PATH . '/fever';
 

--- a/app/Utils/passwordUtil.php
+++ b/app/Utils/passwordUtil.php
@@ -1,7 +1,5 @@
 <?php
 
-declare(strict_types=1);
-
 class FreshRSS_password_Util {
 	// Will also have to be computed client side on mobile devices,
 	// so do not use a too high cost
@@ -20,7 +18,10 @@ class FreshRSS_password_Util {
 		// Compatibility with bcrypt.js
 		$passwordHash = preg_replace('/^\$2[xy]\$/', '\$2a\$', $passwordHash);
 
-		return $passwordHash === '' ? '' : $passwordHash;
+		if ($passwordHash === '' || $passwordHash === null) {
+			return '';
+		}
+		return $passwordHash;
 	}
 
 	/**

--- a/lib/lib_install.php
+++ b/lib/lib_install.php
@@ -1,7 +1,5 @@
 <?php
 
-define('BCRYPT_COST', 9);
-
 Minz_Configuration::register('default_system', join_path(FRESHRSS_PATH, 'config.default.php'));
 Minz_Configuration::register('default_user', join_path(FRESHRSS_PATH, 'config-user.default.php'));
 

